tst/test.gi

```diff
@@ -86,19 +109,25 @@ SOTRec.testIdSOTGroup := function(orders)
 
         sot := List([1..nr],x->SOTGroup(n,x));
         soty := AllSOTGroups(n);
+        signature := SortedList(List(Collected(Factors(n)),x->x[2]));
         for i in [1..nr] do
             Assert(0, HasIdSOTGroup(sot[i]));
             Assert(0, HasIdSOTGroup(soty[i]));
             copies := [];
+            if IsPermGroup(sot[i]) then
+                Assert(0, sot[i] = soty[i]);
+            elif not signature in [ [1,4], [1,1,1,1] ] then # case p^4*q is problematic
+                Assert(0, CodePcGroup(sot[i]) = CodePcGroup(soty[i]));
+            elif CodePcGroup(sot[i]) <> CodePcGroup(soty[i]) then
+                Add(copies, getRandomPc(soty[i]));
```

Comment:
This can be simplified once issue #51 is resolved:
```suggestion
            else
                Assert(0, CodePcGroup(sot[i]) = CodePcGroup(soty[i]));
```